src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbs.tsx

```diff
@@ -71,40 +76,53 @@ class BreadcrumbsContainer extends React.Component<Props, State> {
     }
 
     const breadcrumbTypes: BreadcrumbFilterGroups = [];
-
-    // TODO(Priscila): implement levels
-    //const breadcrumbLevels: BreadcrumbFilterGroups = [];
+    const breadcrumbLevels: BreadcrumbFilterGroups = [];
 
     const convertedBreadcrumbs = breadcrumbs.map((breadcrumb, index) => {
       const convertedBreadcrumb = convertBreadcrumbType(breadcrumb);
-      const breadcrumbDetails = getBreadcrumbDetails(convertedBreadcrumb.type);
+      const breadcrumbTypeDetails = getBreadcrumbTypeDetails(convertedBreadcrumb.type);
 
       if (!breadcrumbTypes.find(b => b.type === convertedBreadcrumb.type)) {
-        !breadcrumbTypes.push({
+        breadcrumbTypes.push({
           groupType: FilterGroupType.TYPE,
           type: convertedBreadcrumb.type,
-          ...breadcrumbDetails,
+          description: breadcrumbTypeDetails.description,
+          symbol: (
+            <BreadcrumbIcon {...omit(breadcrumbTypeDetails, 'description')} size="xs" />
+          ),
+          isChecked: true,
+        });
+      }
+
+      if (!breadcrumbLevels.find(b => b.type === String(convertedBreadcrumb?.level))) {
+        breadcrumbLevels.push({
+          groupType: FilterGroupType.LEVEL,
+          type: String(convertedBreadcrumb?.level) as BreadcrumbLevelType,
```

Comment:
Is this to handle custom levels that users have invented themselves? Won't this cast mislead tsc as BreadcrumbLevelType is an enum?

Reply:
I've done it because I thought that the user might want to filter for undefined levels. I am not expecting to receive another level levels different from the ones I have defined.. anyway.. it's a good point that I am gonna bring to a meeting I will have to talk about the breadcrumbs...thanks Mark.